posts/2020-06-08-announcing-the-moving-from-scala-2-to-scala-3-course.adoc

```diff
@@ -0,0 +1,40 @@
+= Announcing the _Moving from Scala 2 to Scala 3_ course
+eloots
+v1.0, 2020-06-08
+:title: Announcing the Moving from Scala 2 to Scala 3 course
+:tags: [dotty, akka, scala]
+ifdef::backend-html5[]
+:in-between-width: width='85%'
+:half-width: width='50%'
+:half-size:
+:thumbnail: width='60'
+endif::[]
+
+== Moving From Scala 2 to Scala 3
+
+After some early peeks at https://dotty.epfl.ch[the Dotty website] on a few occasions in 2019, I started looking at the https://dotty.epfl.ch/docs/reference/overview.html[Dotty reference documentation] in earnest after reading the release notes of the https://dotty.epfl.ch/blog/2019/12/20/21th-dotty-milestone-release.html[0.21.0-RC1 Dotty release]. That release marked an important milestone, namely, the declaration of Dotty as feature-complete. After reading the article https://www.scala-lang.org/2019/12/18/road-to-scala-3.html[Scala 2 roadmap update: The Road to Scala 3] that was referenced in the release notes, I had all I needed to test some ideas with Dotty's new Union Types. You may have read the full report on this in https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed[part I] and https://www.lunatech.com/2020/02/using-dotty-union-types-with-akka-typed-part-ii[part II] of the ***Using Dotty Union types with Akka Typed*** articles.
+
+Pretty exciting stuff I still think, and I'm sure that this approach will be adopted in a future version of the Akka toolkit as an alternative to the rather convoluted way of dealing with responses to messages in Akka Typed (something which is necessary because of some limitations of the Scala 2 compiler).
```

Comment:
I'd recommend changing this around a bit and not starting the sentence with "Pretty exciting stuff I still think", which reads a bit choppy. I'd recommend still stating what you did about how the Akka toolkit may adopt this as an alternative to the rather convulated current way of doing it, and then say something along the lines of "While I'm still excited about this, I also started wondering about what to do next...", which I think is a natural segue.

Reply:
I'm removing the "Pretty exciting..." part and joining this paragraph with the previous one.